### DISCUSSION TITLE
Add Placeholder

### DISCUSSION THREAD

**Discussion by terryupton** [solspace/craft-express-forms#61] [Feature Requests]
Would it be possible to request  an enhancement of setting the placeholder text on fields?

**Comment by kjmartens** [comment]
Thanks for the feedback, @terryupton. We'll consider this in our next update for Express Forms.

Can I ask you a question for context? Do you plan to / see yourself using the automated rendering template approach, or manually coding your forms?

**Comment by terryupton** [comment]
Hi @kjmartens - I think I would use both to be honest.

For very simple forms  (like a basic enquiry) and the one I have currently set-up on the contact page - I have used the template approach. But If I wanted more control over the layout (columns - fields side by side etc) or had fields like uploads etc, I think I would perhaps opt for a manual approach. 

I have a job enquiry form to also build and I suspect I will do this using the manual approach.

If I had to choose one, then I wouldn't have issues doing all forms and fields manually. I like that level of control for layout.

**Comment by kjmartens** [comment]
Thanks @terryupton. It's not that we would only make the Placeholder available for one approach, but rather that I was just curious how our customers are using/planning to use Express Forms so we have a better understanding for future improvements and direction. 🙂 

When we made specific decisions about whether or not we include features likes placeholder, default value, etc, we made a bit of an assumption that people wanting that level of control would likely be wanting to manually code those values inside the template anyway (thus, making placeholder and default value, etc settings just more clutter in the form builder). No doubt we didn't make perfect decisions, so the feedback you have given us is much appreciated as we consider which improvements to make in the future. :)

**Comment by missmatsuko** [comment]
FYI using placeholder attributes on form inputs is generally considered bad practice:
https://www.smashingmagazine.com/2018/06/placeholder-attribute/
https://www.nngroup.com/articles/form-design-placeholders/
https://medium.com/simple-human/10-reasons-why-placeholders-are-problematic-f8079412b960

An instructions field might be handy, though (if it doesn't exist already).

**Comment by samuelbirch** [comment]
An instruction field would be very useful, also a validation message if the field type is required would be handy too. 

@kjmartens i would generally be manually rendering the fields, but i would like the client to be able to change the instruction text etc.
